src/sources/aws_s3/sqs.rs

```diff
@@ -150,35 +167,75 @@ impl Ingestor {
             multiline,
 
             queue_url: config.queue_url,
-            poll_interval: Duration::from_secs(config.poll_secs),
+            poll_secs: config.poll_secs,
+            client_concurrency: config.client_concurrency,
             visibility_timeout_secs,
             delete_message: config.delete_message,
-        })
+        });
+
+        Ok(Ingestor { state })
     }
 
-    pub(super) async fn run(self, mut out: Pipeline, shutdown: ShutdownSignal) -> Result<(), ()> {
-        let mut stream =
-            IntervalStream::new(time::interval(self.poll_interval)).take_until(shutdown);
+    pub(super) async fn run(self, out: Pipeline, shutdown: ShutdownSignal) -> Result<(), ()> {
+        let mut handles = Vec::new();
+        for _ in 0..self.state.client_concurrency {
+            let process =
+                IngestorProcess::new(Arc::clone(&self.state), out.clone(), shutdown.clone());
+            let handle = tokio::spawn(async move { process.run().await });
+            handles.push(handle);
+        }
 
-        while stream.next().await.is_some() {
-            self.run_once(&mut out).await
+        for handle in handles.drain(..) {
+            if handle.await.is_err() {
+                // TODO: probably emit the error as an internal event but then just return Err(())?
```

Comment:
They _are_ stateless amongst themselves, so theoretically only a drop in throughput and nothing else? It would be odd, though.  I'm not sure if it's better to keep working or to cleanly shut down.  I lean in the "keep working" direction, but it's not a super strong opinion.

Reply:
Yeah, I just mean it'd be an unfortunate failure mode if there's a rare-ish bug that causes a panic in that task, and the symptom is that throughput slowly goes down to zero and then the source shuts down. The logs should make it clear enough what's happening, but I tend to lean towards shutting down immediately when there's evidence some unexpected/unrecoverable has happened.

Reply:
I agree. I think just shutting down will result in more predictable behavior.

Reply:
Addressed in c9f6b35.